de.dashup.model/src/main/java/de/dashup/model/service/DashupService.java

```diff
@@ -268,6 +268,47 @@ public Widget getPanelById(int id) throws SQLException {
         return returningValue;
     }
 
+    public List<Widget> getSimilarWidgets(String category, int currentId) throws SQLException {
+        ArrayList<Widget> returningValue = new ArrayList<>();
+        Map<String, Object> whereParameters = new HashMap<>();
+        whereParameters.put("category", category);
+        List<? extends DatabaseObject> result = this.database.getObject(Database.Table.PANELS, DatabaseWidget.class, whereParameters);
+        if (result != null && result.size() > 3) {
+            for (int i = 0; i < 3; i++) {
```

Comment:
You could shuffle the result list before so that the widgets displayed in the similar section change from time to time if the result set of similar widgets is more than 4.

Reply:
done